File: ruby/cluster_management/lib/delete_multi_region.rb
require 'aws-sdk-core'
require 'aws-sdk-dsql'

def delete_multi_region_cluster(region, arns)
  begin
    # Create client with default configuration and credentials
    client = Aws::DSQL::Client.new(region: region)
    client.delete_multi_region_clusters(
      linked_cluster_arns: arns
    )
  rescue Aws::Errors::ServiceError => e
    raise "Failed to delete multi-region cluster: #{e.message}"
  end
end

File: dotnet/cluster_management/DeleteMultiRegion.cs
using Amazon;
using Amazon.DSQL;
using Amazon.DSQL.Model;
using Amazon.Runtime;

class MultiRegionClusterDeletion {
    public static async Task Delete(RegionEndpoint region, List<string> arns)
    {
        // Create the sdk client
        AWSCredentials aWSCredentials = FallbackCredentialsFactory.GetCredentials();
        AmazonDSQLConfig clientConfig = new()
        {
            AuthenticationServiceName = "dsql",
            RegionEndpoint = region
        };
        AmazonDSQLClient client = new(aWSCredentials, clientConfig);

        // Delete a multi region clusters
        DeleteMultiRegionClustersRequest deleteMultiRegionClustersRequest = new()
        {
            LinkedClusterArns = arns
        };
        DeleteMultiRegionClustersResponse deleteMultiRegionClustersResponse = 
            await client.DeleteMultiRegionClustersAsync(deleteMultiRegionClustersRequest);
        Console.WriteLine(deleteMultiRegionClustersResponse.ResponseMetadata.RequestId);
    }
}
